internal/services/appservice/windows_function_app_resource.go

```diff
@@ -0,0 +1,871 @@
+
+var _ sdk.ResourceWithCustomImporter = WindowsFunctionAppResource{}
+
+var _ sdk.ResourceWithCustomizeDiff = WindowsFunctionAppResource{}
+
+func (r WindowsFunctionAppResource) ModelObject() interface{} {
+	return &WindowsFunctionAppModel{}
+}
+
+func (r WindowsFunctionAppResource) ResourceType() string {
+	return "azurerm_windows_function_app"
+}
+
+func (r WindowsFunctionAppResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.FunctionAppID
+}
+
+func (r WindowsFunctionAppResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.WebAppName,
+			Description:  "Specifies the name of the Function App.",
+		},
+
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"location": location.Schema(),
+
+		"service_plan_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.ServicePlanID,
+			Description:  "The ID of the App Service Plan within which to create this Function App",
+		},
+
+		"storage_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: storageValidate.StorageAccountName,
+			Description:  "The backend storage account name which will be used by this Function App.",
+		},
+
+		"storage_account_access_key": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Sensitive:    true, // TODO - Uncomment this
+			ValidateFunc: validation.NoZeroValues,
+			ExactlyOneOf: []string{
+				"storage_uses_managed_identity",
+				"storage_account_access_key",
+			},
+			Description: "The access key which will be used to access the storage account for the Function App.",
+		},
+
+		"storage_uses_managed_identity": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+			ExactlyOneOf: []string{
+				"storage_uses_managed_identity",
+				"storage_account_access_key",
+			},
+			Description: "Should the Function App use its Managed Identity to access storage",
+		},
+
+		"app_settings": {
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+			Description: "A map of key-value pairs for [App Settings](https://docs.microsoft.com/en-us/azure/azure-functions/functions-app-settings) and custom values.",
+		},
+
+		"auth_settings": helpers.AuthSettingsSchema(),
+
+		"backup": helpers.BackupSchema(),
+
+		"builtin_logging_enabled": {
+			Type:        pluginsdk.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: "Should built in logging be enabled. Configures `AzureWebJobsDashboard` app setting based on the configured storage setting",
+		},
+
+		"client_cert_enabled": {
+			Type:        pluginsdk.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Should the function app use Client Certificates",
+		},
+
+		"client_cert_mode": {
```

Comment:
should we expand cert?
```suggestion
		"client_certificate_enabled": {
			Type:        pluginsdk.TypeBool,
			Optional:    true,
			Default:     false,
			Description: "Should the function app use Client Certificates",
		},

		"client_certificate_mode": {
```

Reply:
As above, this affects all the other resources.